File: src/app/models/public_chat_full.ts
/* models */
import { Chat } from "./chat";
import { User } from "./user";

export class PublicChatFull {
  constructor(
    public id: string,
    public user: User,
    public chat: Chat,
    public createdAt: string,
  ) {}

  public static fromJson(json: any): PublicChatFull {
    return new PublicChatFull(
      json.id,
      json.userId,
      json.chatId,
      json.createdAt,
    );
  }
}

File: src-tauri/src/models/public_chat_full.rs
/* system libraries */
use serde::{Deserialize, Serialize};

/* models */
use super::{chat::Chat, user_data::UserData};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct PublicChatFull {
  pub id: String,
  pub user: UserData,
  pub chat: Chat,
  pub createdAt: String,
}